src/local/BLELocalDevice.h

```diff
@@ -20,12 +20,13 @@
 #ifndef _BLE_LOCAL_DEVICE_H_
 #define _BLE_LOCAL_DEVICE_H_
 
+#include "utility/HCI.h"
 #include "BLEDevice.h"
 #include "BLEService.h"
 
 class BLELocalDevice {
 public:
-  BLELocalDevice();
+  BLELocalDevice(HCITransportInterface *HCITransport);
```

Comment:
Giving the default HCITransport would allow to be compatible with other HCI Transport and kept backward compatibility?
```suggestion
  BLELocalDevice(HCITransportInterface *HCITransport = &HCITransport);
```

Reply:
Actually, with the new approach the HCITransport instance should be defined in the application sketch, so we have more flexibility about the transport that we want to use. I prefer to maintain this parameter as mandatory to be passed at application level

Reply:
Ok